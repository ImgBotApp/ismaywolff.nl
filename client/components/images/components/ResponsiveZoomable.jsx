import React, { PropTypes } from 'react'
import sizeMe from 'react-sizeme'
import Zoomable from './Zoomable'
import { roundUp } from '../utils'

export function ResponsiveZoomable({ id, size }) {
  return (
    <Zoomable
      id={id}
      width={roundUp(size.width)}
    />
  )
}

ResponsiveZoomable.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.object.isRequired
}

export default sizeMe()(ResponsiveZoomable)
