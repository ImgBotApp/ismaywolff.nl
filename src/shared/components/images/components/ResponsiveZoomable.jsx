import React from 'react'
import { string } from 'prop-types'
import { AsyncReactMeasure } from '../../async'
import { getImageWidth } from '../utils'
import Zoomable from './Zoomable'

/* istanbul ignore next: dynamic import won't be executed when testing */
function ResponsiveZoomable({ id }) {
  return (
    <AsyncReactMeasure includeMargin={false} whitelist={['width']}>
      {dimensions => <Zoomable id={id} width={getImageWidth(dimensions.width)} />}
    </AsyncReactMeasure>
  )
}

ResponsiveZoomable.propTypes = {
  id: string.isRequired
}

export default ResponsiveZoomable
