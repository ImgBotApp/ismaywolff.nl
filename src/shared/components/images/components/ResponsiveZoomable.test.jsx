import React from 'react'
import { mount } from 'enzyme'
import ResponsiveZoomable from './ResponsiveZoomable'

jest.mock('./Zoomable', () => () => <div />)

describe('<ResponsiveZoomable />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<ResponsiveZoomable id={'one'} />)
    expect(wrapper).toMatchSnapshot()
  })
})
