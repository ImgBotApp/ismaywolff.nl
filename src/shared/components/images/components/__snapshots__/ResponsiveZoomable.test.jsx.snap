// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveZoomable /> renders correctly 1`] = `
<ResponsiveZoomable
  id="one"
  size={
    Object {
      "width": 50,
    }
  }
>
  <Loadable
    includeMargin={false}
    whitelist={
      Array [
        "width",
      ]
    }
  >
    <Measure
      blacklist={Array []}
      cloneOptions={Object {}}
      includeMargin={false}
      onMeasure={[Function]}
      shouldMeasure={true}
      useClone={false}
      whitelist={
        Array [
          "width",
        ]
      }
    >
      <Component
        id="one"
        width={50}
      >
        <div />
      </Component>
    </Measure>
  </Loadable>
</ResponsiveZoomable>
`;
